### PR TITLE
Improve WebClient upload tests stability

### LINKED ISSUE #36226: Test failure: System.Net.Tests.SyncWebClientTest/UploadData_LargeData_Success(echoServer: http://corefx-net-http11.azurewebsites.net/Echo.ashx)
Opened on behalf of @AriNuer

The test `System.Net.Tests.SyncWebClientTest/UploadData_LargeData_Success(echoServer: http://corefx-net-http11.azurewebsites.net/Echo.ashx)` has failed.

Failure Message:
```
System.Net.WebException : The remote server returned an error: (500) Internal Server Error.
```

Stack Trace:
```
   at System.Net.HttpWebRequest.GetResponse() in /__w/1/s/src/System.Net.Requests/src/System/Net/HttpWebRequest.cs:line 1009
   at System.Net.WebClient.GetWebResponse(WebRequest request) in /__w/1/s/src/System.Net.WebClient/src/System/Net/WebClient.cs:line 249
   at System.Net.WebClient.DownloadBits(WebRequest request, Stream writeStream) in /__w/1/s/src/System.Net.WebClient/src/System/Net/WebClient.cs:line 818
   at System.Net.WebClient.UploadBits(WebRequest request, Stream readStream, Byte[] buffer, Int32 chunkSize, Byte[] header, Byte[] footer) in /__w/1/s/src/System.Net.WebClient/src/System/Net/WebClient.cs:line 982
   at System.Net.WebClient.UploadDataInternal(Uri address, String method, Byte[] data, WebRequest& request) in /__w/1/s/src/System.Net.WebClient/src/System/Net/WebClient.cs:line 451
   at System.Net.WebClient.UploadData(Uri address, String method, Byte[] data) in /__w/1/s/src/System.Net.WebClient/src/System/Net/WebClient.cs:line 435
   at System.Net.WebClient.UploadData(String address, Byte[] data) in /__w/1/s/src/System.Net.WebClient/src/System/Net/WebClient.cs:line 414
   at System.Net.Tests.SyncWebClientTest.<>c__DisplayClass5_0.<UploadDataAsync>b__0() in /__w/1/s/src/System.Net.WebClient/tests/WebClientTest.cs:line 712
   at System.Threading.Tasks.Task`1.InnerInvoke() in /__w/2/s/src/System.Private.CoreLib/shared/System/Threading/Tasks/Future.cs:line 517
   at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj) in /__w/2/s/src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs:line 2439
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state) in /__w/2/s/src/System.Private.CoreLib/shared/System/Threading/ExecutionContext.cs:line 288
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state) in /__w/2/s/src/System.Private.CoreLib/shared/System/Threading/ExecutionContext.cs:line 313
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread) in /__w/2/s/src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs:line 2400
--- End of stack trace from previous location where exception was thrown ---
   at System.Net.Tests.WebClientTestBase.UploadData_LargeData_Success(Uri echoServer) in /__w/1/s/src/System.Net.WebClient/tests/WebClientTest.cs:line 654
--- End of stack trace from previous location where exception was thrown ---
```

Build : 3.0 - 20190321.9 (Core Tests)
Failing configurations:
- Debian.9.Amd64-x64
  - Release
- RedHat.7.Amd64-x64
  - Release
- Ubuntu.1604.Arm64-arm64
  - Release

Details:
https://mc.dot.net/#/product/netcore/30/source/official~2Fdotnet~2Fcorefx~2Frefs~2Fheads~2Fmaster/type/test~2Ffunctional~2Fcli~2F/build/20190321.9/workItem/System.Net.WebClient.Tests/analysis/xunit/System.Net.Tests.SyncWebClientTest~2FUploadData_LargeData_Success(echoServer:%20http:~2F~2Fcorefx-net-http11.azurewebsites.net~2FEcho.ashx)

### PR DESCRIPTION
**Pull request by davidsh**
This PR is a follow-up to PR #36801 which improved HttpClient tests stability.

Some of the WebClient tests were failing with WebExceptions being caused by the
Azure remote endpoint returning 500 status code. Similar to the fixes done to
the HttpClient tests, this PR updates some of the WebClient upload tests to use
the VerifyUpload endpoint which doesn't have to echo back the request body payload.

Some of the WebClient tests upload more complex multi-part formats and can't be
easily changed to use the VerifyUpload endpoint. But those tests aren't using large
data sets so it should be ok.

Closes #36226

### CODE REVIEW

File: src/System.Net.WebClient/tests/WebClientTest.cs [hunk 1]
```diff
@@ -699,6 +707,15 @@ public async Task UploadValues_Success(Uri echoServer)
             byte[] result = await UploadValuesAsync(wc, echoServer.ToString(), new NameValueCollection() { { "Data", ExpectedText } });
             Assert.Contains(ExpectedTextAfterUrlEncode, Encoding.UTF8.GetString(result));
         }
+
+        private static void AddMD5Header(WebClient wc, string data)
```
Comment (rmkerr) [inline src/System.Net.WebClient/tests/WebClientTest.cs]: Nit: I think the implicit conversion to UTF-8 in this method is potentially confusing. Maybe change the name, or just add a comment to make that clear?

Comment (davidsh) [reply src/System.Net.WebClient/tests/WebClientTest.cs]: Thx.  I'll add a comment to clarify the use of UTF-8.

**Pull request merged**